core/src/main/java/hudson/model/ItemGroup.java

```diff
@@ -88,4 +89,42 @@ default void onRenamed(T item, String oldName, String newName) throws IOExceptio
      * Internal method. Called by {@link Item}s when they are deleted by users.
      */
     void onDeleted(T item) throws IOException;
+
+    /**
+     * Gets all the {@link Item}s recursively in the {@link ItemGroup} tree
+     * and filter them by the given type.
+     * @since FIXME
+     */
+    default <T extends Item> List<T> getAllItems(Class<T> type) {
+        return Items.getAllItems(this, type);
+    }
+
+    /**
+     * Gets all the {@link Item}s unordered, lazily and recursively in the {@link ItemGroup} tree
+     * and filter them by the given type.
+     * @since FIXME
+     */
+    default <T extends Item> Iterable<T> allItems(Class<T> type) {
+        return Items.allItems(this, type);
+    }
+
+    /**
+     * Gets all the items recursively.
+     * @since FIXME
+     */
+    default List<Item> getAllItems() {
+        return getAllItems(Item.class);
+    }
+
+    /**
+     * Gets all the items unordered, lazily and recursively.
+     * @since FIXME
+     */
+    default Iterable<Item> allItems() {
+        return allItems(Item.class);
+    }
+
+    // TODO could delegate to allItems overload taking Authentication, but perhaps more useful to introduce a variant to perform preauth filtering using Predicate and check Item.READ afterwards
+    // or return a Stream<Item> and provide a Predicate<Item> public static Items.readable(), and see https://stackoverflow.com/q/22694884/12916 if you are looking for just one result
```

Comment:
Could easily write such an impl, but do not have a candidate caller handy to use as a downstream PR, and could find no cases of this idiom in core. (https://github.com/jenkinsci/active-choices-plugin/pull/17 could use it, I was just too lazy to file a PR based on a snapshot baseline for it. Maybe if someone expresses interest.)